### ISSUE TITLE
[berkeley-qa] exception on handling REST server request

### ISSUE THREAD

**Issue by p-shahi** [MinaProtocol/mina#10955]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/MinaProtocol/mina/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/MinaProtocol/mina/discussions

### Description

```
2022-05-13 16:47:39 UTC [Error] Exception while handling REST server request: $error
        error: "(monitor.ml.Error(\"Fields_derivers_zkapps.Make(Schema).Invalid_rich_scalar(-608348572)\")(\"Raised at Fields_derivers_zkapps.proof.of_string in file \\\"src/lib/fields_derivers_zkapps/fields_derivers_zkapps.ml\\\", line 448, characters 8-42\"\"Called from Base__Option.map in file \\\"src/option.ml\\\", line 222, characters 23-28\"\"Called from Mina_base__Control.As_record.Fields.make_creator.(fun) in file \\\"src/lib/mina_base/control.ml\\\", line 99, characters 7-8\"\"Called from Graphql_schema.Make.Arg.eval_arglist.(fun) in file \\\"graphql/src/graphql_schema.ml\\\", line 277, characters 84-95\"\"Called from Rresult.R.bind in file \\\"src/rresult.ml\\\" (inlined), line 30, characters 38-41\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-213\"\"Called from Rresult.R.bind in file \\\"src/rresult.ml\\\" (inlined), line 30, characters 38-41\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-213\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 309, characters 16-90\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 323, characters 12-77\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-98\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 309, characters 16-90\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 323, characters 12-77\"\"Called from Stdlib__list.map in file \\\"list.ml\\\", line 92, characters 20-23\"\"Called from Graphql_schema.List.Result.all in file \\\"graphql/src/graphql_schema.ml\\\", line 17, characters 6-19\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 317, characters 16-107\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 323, characters 12-77\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-98\"\"Called from Rresult.R.bind in file \\\"src/rresult.ml\\\" (inlined), line 30, characters 38-41\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-213\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 309, characters 16-90\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 323, characters 12-77\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-98\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 309, characters 16-90\"\"Called from Graphql_schema.Make.Arg.eval_arg in file \\\"graphql/src/graphql_schema.ml\\\", line 323, characters 12-77\"\"Called from Graphql_schema.Make.Arg.eval_arglist in file \\\"graphql/src/graphql_schema.ml\\\", line 276, characters 14-98\"\"Called from Graphql_schema.Make.resolve_field in file \\\"graphql/src/graphql_schema.ml\\\", line 1383, characters 12-107\"\"Called from Graphql_schema.Make.Io.map_s in file \\\"graphql/src/graphql_schema.ml\\\", line 67, characters 15-20\"\"Called from Graphql_schema.Make.resolve_fields in file \\\"graphql/src/graphql_schema.ml\\\", line 1408, characters 6-580\"\"Called from Graphql_schema.Make.execute_operation.(fun) in file \\\"graphql/src/graphql_schema.ml\\\", line 1497, characters 15-74\"\"Called from Async_kernel__Deferred0.bind.(fun) in file \\\"src/deferred0.ml\\\", line 54, characters 64-69\"\"Called from Async_kernel__Job_queue.run_job in file \\\"src/job_queue.ml\\\" (inlined), line 128, characters 2-5\"\"Called from Async_kernel__Job_queue.run_jobs in file \\\"src/job_queue.ml\\\", line 169, characters 6-47\"\"Caught by monitor Cohttp_async.Server.collect_errors\"))"
```

reported by @garethtdavies 

### Steps to Reproduce

Deploying a zkApp and sending a transaction with a modified version of this script:  https://github.com/o1-labs/snarkyjs/blob/main/src/examples/simple_zkapp_berkeley.ts

Currently waiting on community member to provide the exact script/steps to repro

### Expected Result

No error

### Actual Result

Exception raised
```
(monitor.ml.Error(\"Fields_derivers_zkapps.Make(Schema).Invalid_rich_scalar(-608348572)\")(\"Raised at Fields_derivers_zkapps.proof.of_string in file \\\"src/lib/fields_derivers_zkapps/fields_derivers_zkapps.ml
```

### How frequently do you see this issue?

Sometimes

### What is the impact of this issue on your ability to run a node?

Low

### Status

```shell
none
```

**Comment by arvidsaur** [comment]
The only steps to reproduce are to run the script twice with a new private key for the zkApp. The first run is to deploy the zkApp and the second run will update the zkApp as the script see that it exists. This script used to work as-is before the recently re-deploy, but now it will successfully deploy the app but the second run will no longer update it.

**Comment by deepthiskumar** [comment]
@arvidsaur could you paste the transaction that fails?

**Comment by p-shahi** [comment]
This issue was due to an incompatibility between the deployed Berkeley QA version and the released snarkyJS version. Closing since this incompatibility has been addressed
